File: linear_search/python/range_search.py
def search_range(array, range_value, target):
    for index in range(range_value[0], range_value[1]+1):
        if array[index] == target:
            return index
    
    return -1


array = [1,2,4,6,33,4,5]
range_value = [1,5]
target = 4

ans = search_range(array, range_value, target)
print(ans)

File: linear_search/java/range_search.java
package linear_search.java;

// Lets say that we have given an array and we have to search within a range
public class range_search {
    public static void main(String[] args){
        int[] nums = {23,45,1,2,3,6,9,1,10};
        int[] range = {1,4};
        int target = 2;
        int ans = range_search(nums, range, target);
        System.out.println(ans);
    }


    static int range_search(int[] nums, int[] range, int target){
        for(int index = range[0]; index<=range[1]; index++){
            if(nums[index]==target){
                return index;
            }
        }
        return -1;
    }
}
